Friendly-plans/app/src/androidTest/java/pg/autyzm/friendly_plans/manager_app/ChildListActivityTest.java

```diff
@@ -122,4 +124,84 @@ public void whenChildIsAddedToDBExpectProperlyDisplayedOnRecyclerView() {
+        final int testedChildPosition = 0;
+        final int firstPosition = 0;
+        onView(withId(R.id.menu_search)).perform(typeText(EXPECTED_FIRST_NAME));
+        onView(withRecyclerView(R.id.rv_child_list)
+                .atPosition(firstPosition))
+                .check(matches(hasDescendant(withText(EXPECTED_FIRST_NAME + " " + EXPECTED_LAST_NAME
+                        + testedChildPosition))));
+    }
+
+    @Test
+    public void whenSearchForAChildWithOneLetterOfSurnameExpectThatChildOnFirstPosition() {
+        final int testedChildPosition = 0;
+        final int firstPosition = 0;
+        onView(withId(R.id.menu_search)).perform(typeText(Character.toString(EXPECTED_LAST_NAME.charAt(0))));
+        onView(withRecyclerView(R.id.rv_child_list)
+                .atPosition(firstPosition))
+                .check(matches(hasDescendant(withText(EXPECTED_FIRST_NAME + " " + EXPECTED_LAST_NAME
+                        + testedChildPosition))));
+    }
+    @Test
+    public void whenSearchForAChildWithSurnameExpectThatChildOnFirstPosition(){
+        final int testedChildPosition = 3;
+        final int firstPosition = 0;
+        onView(withId(R.id.menu_search)).perform(typeText(EXPECTED_LAST_NAME + testedChildPosition));
+        onView(withRecyclerView(R.id.rv_child_list)
+                .atPosition(firstPosition))
+                .check(matches(hasDescendant(withText(EXPECTED_FIRST_NAME + " " + EXPECTED_LAST_NAME
+                        + testedChildPosition))));
+    }
+    @Test
+    public void whenSearchForAChildWithNameAndSurnameExpectOnlyThatChildToAppear() { //TO DO
+        final int testedChildPosition = 3;
+        final int firstPosition = 0;
+        final int nextPosition = 1;
+        onView(withId(R.id.menu_search)).perform(typeText(
+                EXPECTED_FIRST_NAME + " " + EXPECTED_LAST_NAME + testedChildPosition));
+        onView(withRecyclerView(R.id.rv_child_list)
+                .atPosition(firstPosition))
+                .check(matches(hasDescendant(withText(EXPECTED_FIRST_NAME + " " + EXPECTED_LAST_NAME
+                        + testedChildPosition))));
+        onView(withRecyclerView(R.id.rv_plan_list)
+                .atPosition(nextPosition)).check(doesNotExist());
+    }
+    @Test
+    public void whenSearchForAChildWithSurnameAndNameExpectOnlyThatChildToAppear(){
+        final int testedChildPosition = 3;
+        final int firstPosition = 0;
+        onView(withId(R.id.menu_search)).perform(typeText(
+                EXPECTED_LAST_NAME + testedChildPosition + " " + EXPECTED_FIRST_NAME));
+        onView(withRecyclerView(R.id.rv_child_list)
+                .atPosition(firstPosition))
+                .check(matches(hasDescendant(withText(EXPECTED_FIRST_NAME + " " + EXPECTED_LAST_NAME
+                        + testedChildPosition))));
+    }
+    @Test
+    public void whenSearchForAChildWithFirstLettersOfNameAndSurnameExpectThatChildOnFirstPosition(){
```

Comment:
As I understand the tests search by first letter of surname and first of name eg. John Smith will be found after J S . I don't think this case should be covered. I think part of name or surname or both is enough. What do you think ?